internal/supervisor/installation.go

```diff
@@ -32,6 +33,13 @@ type installationProvisioner interface {
 	CreateClusterInstallation(cluster *model.Cluster, installation *model.Installation, clusterInstallation *model.ClusterInstallation) error
 	DeleteClusterInstallation(cluster *model.Cluster, installation *model.Installation, clusterInstallation *model.ClusterInstallation) error
 	UpdateClusterInstallation(cluster *model.Cluster, installation *model.Installation, clusterInstallation *model.ClusterInstallation) error
+	GetClusterInstallationResource(cluster *model.Cluster, installation *model.Installation, clusterInstallation *model.ClusterInstallation) (*mmv1alpha1.ClusterInstallation, error)
+}
+
+// aws abstracts the aws client operations required by the installation supervisor.
```

Comment:
nit: it would nice to converge on a name that doesn't couple us to a particular cloud provider (for that mythical future ;P). Maybe `registrar` (meh) or `dnsManager`?

Reply:
This is a good point. I am not entirely happy with how I mocked out the code for testing anyway, but haven't come up with anything better yet. I will think on it and may do a small follow-up PR to make some improvements.